src/legacy/core_plugins/kbn_vislib_vis_types/public/components/common/color_schema.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiLink, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { VisOptionsProps } from 'ui/vis/editors/default';
+import { ColorSchema } from 'ui/vislib/components/color/colormaps';
+import { SelectOption } from './select';
+import { SwitchOption } from './switch';
+import { ColorSchemaVislibParams } from '../../types';
+
+export type SetColorSchemaOptionsValue = <T extends keyof ColorSchemaVislibParams>(
+  paramName: T,
+  value: ColorSchemaVislibParams[T]
+) => void;
+
+interface ColorSchemaOptionsProps extends ColorSchemaVislibParams {
+  disabled?: boolean;
+  colorSchemas: ColorSchema[];
+  uiState: VisOptionsProps['uiState'];
+  setValue: SetColorSchemaOptionsValue;
+}
+
+function ColorSchemaOptions({
+  disabled,
+  colorSchema,
+  colorSchemas,
+  invertColors,
+  uiState,
+  setValue,
+}: ColorSchemaOptionsProps) {
+  const [isCustomColors, setIsCustomColors] = useState(false);
```

Comment:
```suggestion
  const [isCustomColors, setIsCustomColors] = useState(() => !!uiState.get('vis.colors'));
```

Could you please add this check for set individual colors? This will fix #43666.

Reply:
You are right, thanks. I would just move it into a separate PR (as we have a separate feature request for this), but if you think it would be better to do it here..